### PR TITLE
Fix miniconda path issue #26962

### LINKED ISSUE #26962: CI: Fix Azure Pipelines problem with PATH
We should be able to set the environment variable `PATH` in an Azure Pipelines job with `echo '##vso[task.prependpath]$HOME/miniconda3/bin'`. See https://github.com/pandas-dev/pandas/blob/master/azure-pipelines.yml#L27

But that didn't work when we implemented it, so we ended up setting the path manually in every step (e.g. https://github.com/pandas-dev/pandas/blob/master/azure-pipelines.yml#L48).

This is adding noise to our CI configuration, making it harder to see what every step is doing. We should try to understand what's the problem with setting the `PATH`, and fix it.

This also applies to `ci/azure/posix.yml` (besides `azure-pipelines.yml` in the root).

### PR DESCRIPTION
**Pull request by bhavaniravi**
- [x] closes #26962
- [ ] tests added / passed
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Review by gfyoung — APPROVED** [review]
Nice!

@datapythonista : If you have anything to add here

File: azure-pipelines.yml [hunk 1]
```diff
@@ -24,78 +24,68 @@ jobs:
   - script: |
       # XXX next command should avoid redefining the path in every step, but
       # made the process crash as it couldn't find deactivate
```
Comment (datapythonista) [inline azure-pipelines.yml]: the comment is not relevant anymore, you fixed it :)

File: azure-pipelines.yml [hunk 2]
```diff
@@ -24,78 +24,68 @@ jobs:
...
-      ci/code_checks.sh patterns
+  - script: ci/code_checks.sh patterns
     displayName: 'Looking for unwanted patterns'
     condition: true
 
   - script: |
-      export PATH=$HOME/miniconda3/bin:$PATH
       sudo apt-get install -y libc6-dev-i386
       ci/setup_env.sh
+      echo "$PATH"
```
Comment (datapythonista) [inline azure-pipelines.yml]: do we want this, or was it for debugging? I guess was for debugging, if that's the case, can you please remove it

File: ci/azure/posix.yml [hunk 3]
```diff
@@ -56,17 +56,16 @@ jobs:
...
     - script: |
         if [ "$(uname)" == "Linux" ]; then sudo apt-get install -y libc6-dev-i386 $EXTRA_APT; fi
+        echo '##vso[task.prependpath]$(HOME)/miniconda3/bin'
         echo "Creating Environment"
         ci/setup_env.sh
       displayName: 'Setup environment and build pandas'
     - script: |
-        export PATH=$HOME/miniconda3/bin:$PATH
         source activate pandas-dev
+        echo "$PATH"
```
Comment (datapythonista) [inline ci/azure/posix.yml]: same as before

**Review by datapythonista — COMMENTED** [review]
really nice clean up, thanks for taking care of this @bhavaniravi 

just couple of minor things, but looks great

**Comment by datapythonista** [comment]
Thanks @bhavaniravi, really great to have this fixed, much cleaner now

**Comment by bhavaniravi** [comment]
Thank you @datapythonista I learned a lot in the process. Looking to contribute more on the devops side of things

File: ci/azure/windows.yml [hunk 4]
```diff
@@ -17,7 +17,9 @@ jobs:
         CONDA_PY: "37"
 
   steps:
-    - powershell: Write-Host "##vso[task.prependpath]$env:CONDA\Scripts"
+    - powershell: |
+        Write-Host "##vso[task.prependpath]$env:CONDA\Scripts"
+        Write-Host "##vso[task.prependpath]$HOME/miniconda3/bin"
```
Comment (jbrockmendel) [inline ci/azure/windows.yml]: @bhavaniravi  any idea if there is a way to do this without powershell?  now that windows supports bash, im trying to see if we can get rid of windows-specific stuff

Comment (datapythonista) [reply ci/azure/windows.yml]: In bash this is a simple echo I think.

Comment (jbrockmendel) [reply ci/azure/windows.yml]: Thats what I thought too, see #30393 where I tried that in the first commit and reverted it when that failed

**Pull request merged**